tools/configen/configen/configen.py

```diff
@@ -128,28 +129,23 @@ def is_incompatible(type_: Type[Any]) -> bool:
     return True
 
 
-def get_default_flags(module: ModuleConf) -> List[Parameter]:
-    flag_type_dict = {
-        "_convert_": "str",
-        "_recursive_": "bool",
-    }
+def get_default_flags(module: ModuleConf, imports_set) -> List[Parameter]:
 
     def_flags: List[Parameter] = []
-    for name, default in module.default_flags.items():
-        try:
-            flag_type = flag_type_dict[name]
-        except KeyError:
-            log.error(f"Flag '{name}' unsupported by configen.")
-            raise
-        default = f'"{default}"' if flag_type == "str" else default
-
-        def_flags.append(
-            Parameter(
-                name=name,
-                type_str=flag_type,
-                default=default,
+    if module.default_flags:
+        for name, default in module.default_flags.items():
+            flag_type = Flags.__annotations__[name].__args__[0].__name__
+
+            if name == "_convert_":
+                imports_set.add(ConvertMode)
```

Comment:
I believe the code can handle both string and ConvertMode as the values for convert.
so far, the generated dataclasses did not depend on Hydra. let's keep it this way and use a string for the _convert_ type in the generated code.

Reply:
done